exporter/awsemfexporter/bounded_pusher_map.go

```diff
@@ -0,0 +1,76 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package awsemfexporter
+
+import (
+	"errors"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/cwlogs"
+)
+
+const (
+	pusherMapLimit = 1000
+)
+
+type BoundedPusherMap struct {
+	pusherMap          map[string]cwlogs.Pusher
+	limit              int
+	stalePusherTracker map[string]time.Time
+}
+
+func NewBoundedPusherMap() BoundedPusherMap {
+	return BoundedPusherMap{
+		pusherMap:          make(map[string]cwlogs.Pusher),
+		limit:              pusherMapLimit,
+		stalePusherTracker: make(map[string]time.Time),
+	}
+}
+
+func (bpm *BoundedPusherMap) Add(key string, pusher cwlogs.Pusher, logger *zap.Logger) {
+	err := bpm.EvictStalePushers()
+	if err != nil {
+		logger.Error("Error with evicting stale pushers", zap.Error(err))
+		return
+	}
+	bpm.pusherMap[key] = pusher
+	bpm.stalePusherTracker[key] = time.Now()
+}
+
+func (bpm *BoundedPusherMap) Get(key string) (cwlogs.Pusher, bool) {
+	pusher, ok := bpm.pusherMap[key]
+	if ok {
+		bpm.stalePusherTracker[key] = time.Now()
+	}
+	return pusher, ok
+}
+
+func (bpm *BoundedPusherMap) EvictStalePushers() error {
+	if len(bpm.pusherMap) < bpm.limit {
+		return nil
+	}
+	now := time.Now()
+	for key, lastUsed := range bpm.stalePusherTracker {
```

Comment:
Can you add a comment on what basis we are deleting the pushermap and stale push tracker.